src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp

```diff
@@ -525,5 +526,41 @@ void SubgraphBaseTest::init_input_shapes(const std::vector<InputShape>& shapes)
         }
     }
 }
+
+ElementType SubgraphBaseTest::get_default_imp_precision_type(ElementType type) {
+#if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
+    return type;
+#endif
+#if defined(OPENVINO_ARCH_RISCV64)
+    return type;
+#endif
+#if defined(OPENVINO_ARCH_X86_64)
+    const std::string key = ov::hint::inference_precision.name();
+    const std::string KEY_ENFORCE_BF16 = "ENFORCE_BF16";
+    // if is not float
+    if (type != ElementType::f16 && type != ElementType::f32 && type != ElementType::bf16) {
+        return type;
+    }
+    // ngraph tranform stage
+    if (type == ElementType::bf16) {
+        type = ov::with_cpu_x86_avx512_core() ? ElementType::bf16 : ElementType::f32;
+    } else {
```

Comment:
why these is no `else if (type == ElementType::f16)` case ?

Reply:
because f16 will be always transformed to f32, https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L292